### PR TITLE
vuln.yml with latest and stable Go version compatibility

### LINKED ISSUE #2573: .github/workflows: use `https://go.dev/dl/?mode=json` to get Go versions for vuln.yml
### Operating System

- [ ] Windows
- [ ] macOS
- [ ] Linux
- [ ] FreeBSD
- [ ] OpenBSD
- [ ] Android
- [ ] iOS
- [ ] Nintendo Switch
- [ ] Xbox
- [ ] Web Browsers

### What feature would you like to be added?

https://stackoverflow.com/questions/46623153/how-do-download-the-latest-version-of-go

> The problem with using curl https://golang.org/VERSION?m=text is that on some days, it will give "bleeding edge" result that is not actually available for download. Like today 2022-04-12 (: It returns 1.18.1 which is not yet available for download.

### Why is this needed?

_No response_

### PR DESCRIPTION
**Pull request by nitinmewar**
Signed-off-by: nitinmewar [nitinmewar28@gmail.com](mailto:nitinmewar28@gmail.com)

# What issue is this addressing?

Closes #2573 

## What _type_ of issue is this addressing?

feature - get stable go versions for vuln.yml

### CODE REVIEW

**Comment by nitinmewar** [comment]
@hajimehoshi .

File: .github/workflows/vuln.yml [hunk 1]
```diff
@@ -2,12 +2,15 @@ name: Vulnerability Check
...
+env:
+  GO_VERSIONS: ${{ steps.get_go_version.outputs.versions }}
+
 jobs:
   test:
     strategy:
       matrix:
         os: [ubuntu-latest, macos-latest, windows-latest]
-        go: ['1.19.6', '1.20.1']
+        go: ${{ env.GO_VERSIONS }}
```
Comment (hajimehoshi) [inline .github/workflows/vuln.yml:13]: 1. GO_VERSIONS is determined after `get_go_version.sh` is executed, right?
2. GO_VERSIONS will be like `go1.19.6`, not `1.19.6`
3. This doesn't work when the version is for example `1.21`. actions/setup-go@v3 requires `1.21.0` in this case.

**Comment by hajimehoshi** [comment]
https://github.com/hajimehoshi/ebiten/actions/runs/4199772403

> The workflow is not valid. .github/workflows/vuln.yml (Line: 6, Col: 16): Unrecognized named-value: 'steps'. Located at position 1 within expression: steps.get_go_version.outputs.versions .github/workflows/vuln.yml (Line: 13, Col: 13): Unrecognized named-value: 'env'. Located at position 1 within expression: env.GO_VERSIONS

File: .github/workflows/vuln.yml (hunk 1 above)
Comment (nitinmewar) [reply .github/workflows/vuln.yml:13]: 1. yes!
2. changing it right now. will do a commit soon.

**Comment by nitinmewar** [comment]
> https://github.com/hajimehoshi/ebiten/actions/runs/4199772403
> 
> > The workflow is not valid. .github/workflows/vuln.yml (Line: 6, Col: 16): Unrecognized named-value: 'steps'. Located at position 1 within expression: steps.get_go_version.outputs.versions .github/workflows/vuln.yml (Line: 13, Col: 13): Unrecognized named-value: 'env'. Located at position 1 within expression: env.GO_VERSIONS

is it because my `get_go_versions.sh` should be sourced first? or something else?

**Comment by hajimehoshi** [comment]
Still the same error.

> is it because my get_go_versions.sh should be sourced first? or something else?

TBH I don't know. Please try it with your repository before updating this PR.

**Comment by hajimehoshi** [comment]
Let me close this.

**Pull request closed without merging**